### PR TITLE
[1] feat(653): add blockedBy

### PR DESCRIPTION
**Pull request by d2lam**
Allow `blockedBy` to job config. It's an array of job names, similar to `requires`
Also add `blockedBy` to executor `start`. This will be an array of job Ids. 

Related: https://github.com/screwdriver-cd/screwdriver/issues/653

### CODE REVIEW

File: config/job.js [hunk 1]
```diff
@@ -67,10 +67,10 @@ const SCHEMA_STEPS = Joi.array().items(SCHEMA_STEP).min(1);
 const SCHEMA_TEMPLATE = Joi.string().regex(Regex.FULL_TEMPLATE_NAME);
 const SCHEMA_JOBNAME = Joi.string().regex(Regex.JOB_NAME);
 const SCHEMA_TRIGGER = Joi.string().regex(Regex.TRIGGER); // ~commit, ~pr, etc.
-const SCHEMA_REQUIRES_VALUE = Joi.alternatives().try(SCHEMA_JOBNAME, SCHEMA_TRIGGER);
-const SCHEMA_REQUIRES = Joi.alternatives().try(
-    Joi.array().items(SCHEMA_REQUIRES_VALUE),
-    SCHEMA_REQUIRES_VALUE
+const SCHEMA_REQUIRES_OR_BLOCKEDBY_VALUE = Joi.alternatives().try(SCHEMA_JOBNAME, SCHEMA_TRIGGER);
```
Comment (tkyi) [inline config/job.js]: Does this mean you can configure `blockedBy: [~pr, ~commit]`?

Comment (d2lam) [reply config/job.js]: Shouldn't. Let me fix it.

File: config/job.js [hunk 2]
```diff
@@ -72,6 +72,7 @@ const SCHEMA_REQUIRES = Joi.alternatives().try(
     Joi.array().items(SCHEMA_REQUIRES_VALUE),
     SCHEMA_REQUIRES_VALUE
 );
+const SCHEMA_BLOCKEDBY = Joi.array().items(SCHEMA_JOBNAME);
```
Comment (tkyi) [inline config/job.js]: Wondering if we will support a single item or an array; also can blockedBy values be external triggers?

File: config/job.js [hunk 3]
```diff
@@ -67,11 +67,14 @@ const SCHEMA_STEPS = Joi.array().items(SCHEMA_STEP).min(1);
...
 const SCHEMA_JOBNAME = Joi.string().regex(Regex.JOB_NAME);
 const SCHEMA_TRIGGER = Joi.string().regex(Regex.TRIGGER); // ~commit, ~pr, etc.
+const SCHEMA_EXTERNAL_TRIGGER = Joi.string().regex(Regex.EXTERNAL_TRIGGER);
 const SCHEMA_REQUIRES_VALUE = Joi.alternatives().try(SCHEMA_JOBNAME, SCHEMA_TRIGGER);
 const SCHEMA_REQUIRES = Joi.alternatives().try(
     Joi.array().items(SCHEMA_REQUIRES_VALUE),
     SCHEMA_REQUIRES_VALUE
 );
+const SCHEMA_BLOCKEDBY_VALUE = Joi.alternatives().try(SCHEMA_JOBNAME, SCHEMA_EXTERNAL_TRIGGER);
+const SCHEMA_BLOCKEDBY = Joi.array().items(SCHEMA_BLOCKEDBY_VALUE);
```
Comment (tkyi) [inline config/job.js]: Should this also be 
```
const SCHEMA_BLOCKEDBY = Joi.alternatives().try(
     Joi.array().items(SCHEMA_BLOCKEDBY_VALUE),
	 SCHEMA_BLOCKEDBY_VALUE
);	
```

**Comment by d2lam** [comment]
To address Pranav's comment in the worker (https://github.com/screwdriver-cd/queue-worker/pull/60#pullrequestreview-120466218), this now also pass the `jobId`. Please re-review the last commit. :d2thanks

**Pull request merged**